qiskit/transpiler/passes/layout/vf2_post_layout.py

```diff
@@ -222,10 +221,21 @@ def run(self, dag):
             # mode the node matcher will not match since none of the circuit ops
             # will match the cmap ops.
             if not self.strict_direction:
-                has_operations = set(itertools.chain.from_iterable(self.target.qargs))
-                to_remove = set(cm_graph.node_indices()).difference(has_operations)
+                has_none = False
+            has_operations = set()
+            for qarg in self.target.qargs:
+                if qarg is None:
+                    has_none = True
+                else:
+                    has_operations.update(qarg)
+            # If there are defined operations or if has_operations is empty and there are
+            # no globally defined gates. If has_none was true and there are no operations
+            # defined that means all operations are supported on all qubits and there is
+            # nothing to remove.
+            if has_operations or not has_none:
+                to_remove = set(range(len(cm_nodes))).difference(has_operations)
                 if to_remove:
-                    cm_graph.remove_nodes_from(list(to_remove))
+                    cm_graph.remove_nodes_from([cm_nodes[i] for i in to_remove])
```

Comment:
I just realized this was a bad copy and paste from vf2 layout. I'll need to update it and I'll add the test I forgot to commit that covers vf2post layout too.

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/10633/commits/8b42060383782c33eed48a9e93f41e9f4418cd5f